### PR TITLE
fix(config): Move normalization config to top level

### PR DESCRIPTION
**Pull request by iker-barriocanal**
This PR unblocks the roll out of enabling normalization only in PoP Relays.

Normalization config was inside the `processing` config, which requires specific config for processing Relays. This is convenient as it raises errors during Relay startup when core features are not properly configured. However, normalization config lives inside the processing one and it's not possible to enable it in non-processing relays. This PR moves it to a separate top-level config to make that possible.

The last Relay release, on March 15, did not include the broken config, so it's ok to assume the config isn't used.

#skip-changelog

### CODE REVIEW

File: tests/integration/test_store.py [hunk 1]
```diff
@@ -1320,30 +1320,30 @@ def test_kafka_ssl(relay_with_processing):
...
     relay,
-    normalize,
+    normalization_level,
 ):
     project_id = 42
     mini_sentry.add_basic_project_config(project_id)
     relay = relay(
         upstream=mini_sentry,
-        options={"processing": {"normalize": normalize}},
+        options={"normalization": {"level": normalization_level}},
```
Comment (iker-barriocanal) [inline tests/integration/test_store.py:1335]: The test should have failed with invalid config -- https://github.com/getsentry/relay/issues/3429.

**Review by iker-barriocanal — COMMENTED** [review]
Open to better suggestions for config names 😄

**Review by Dav1dde — APPROVED** [review]
> The last Relay release, on March 15, did not include the broken config, so it's ok to assume the config isn't used.

The last release was yesterday (unlucky), but still fine to assume that.

File: relay-config/src/config.rs [hunk 2]
```diff
@@ -1035,23 +1032,31 @@ impl Default for Processing {
...
+#[derive(Debug, Default, Serialize, Deserialize)]
+#[serde(default)]
+pub struct Normalization {
+    /// Level of normalization for Relay to apply to incoming data.
+    #[serde(default)]
+    pub level: NormalizationLevel,
+}
+
 /// Configuration for the level of normalization this Relay should do.
 #[derive(Clone, Debug, Default, Serialize, Deserialize, Eq, PartialEq)]
```
Comment (Dav1dde) [inline relay-config/src/config.rs]:
```suggestion
#[derive(Copy, Clone, Debug, Default, Serialize, Deserialize, Eq, PartialEq)]
```

File: relay-config/src/config.rs [hunk 3]
```diff
@@ -2188,9 +2195,9 @@ impl Config {
         self.values.processing.enabled
     }
 
-    /// How much normalization this Relay should apply to incoming data.
-    pub fn normalization(&self) -> &Normalize {
-        &self.values.processing.normalize
+    /// Level of normalization for Relay to apply to incoming data.
+    pub fn normalization_level(&self) -> &NormalizationLevel {
+        &self.values.normalization.level
```
Comment (Dav1dde) [inline relay-config/src/config.rs]:
```suggestion
    pub fn normalization_level(&self) -> NormalizationLevel {
        self.values.normalization.level
```

**Review by Dav1dde — COMMENTED** [review]
Okay, 2 mini nits.

**Pull request merged**